src/ITensors.jl

```diff
@@ -182,3 +182,17 @@ end
 # TODO: make this definition AbstractMPS
 # Handle orthogonality center correctly
 Base.getindex(ψ::MPS, r::UnitRange{Int}) = MPS([ψ[n] for n in r])
+
+# Base.zero(T::ITensor) = false * T
+#
+# function ITensors.NDTensors.datatype(
+#   ::ITensors.EmptyStorage{Float64,ITensors.NDTensors.BlockSparse{Float64,Vector{Float64},N}}
+# ) where {N}
+#   return Vector{Float64}
+# end
+#
+# function ITensors.NDTensors.datatype(T::ITensors.NDTensors.TensorStorage{Float64})
+#   return typeof(data(T))
+# end
+
+Base.fill!(::NDTensors.NoData, ::Any) = 0
```

Comment:
This is definition seems a bit funny, is this needed?

Reply:
I forgot this was fixing the issue with `zero` of an `EmptyStorage`.

It looks like this is sufficient:
```julia
fill!(::NoData, ::EmptyNumber) = NoData()
```
and is a bit safer/more specific. I'll add that to NDTensors.

Reply:
I had to keep that in (with the NoData()), it seems the newest version of ITensors does not yet have it?